app/scripts/modules/core/filterModel/FilterTags.tsx

```diff
@@ -13,11 +11,65 @@ export interface IFilterTag extends IFilter {
   clear: () => any;
 }
 
-interface IProps {
+export interface IProps {
   tags: IFilterTag[];
   tagCleared?: () => any;
   clearFilters: () => any;
 }
 
-export let FilterTags: React.ComponentClass<IProps> = undefined;
-ReactInjector.give(($injector: any) => FilterTags = angular2react('filterTags', filterTagsComponent, $injector) as any);
+export interface IState {
+  tags: IFilterTag[];
+}
+
+@autoBindMethods
+export class FilterTags extends React.Component<IProps, IState> {
+
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      tags: props.tags
+    };
+  }
+
+  public componentWillReceiveProps(newProps: IProps) {
+    this.setState({ tags: newProps.tags });
+  }
+
+  private clearAllFilters(): void {
+    this.props.clearFilters();
+    ReactGA.event({category: 'Filter Tags', action: 'Clear All clicked'});
+  }
+
+  private generateTag(tag: IFilterTag) {
+    const clearFilter = (): void => {
+      tag.clear();
+      if (this.props.tagCleared) {
```

Comment:
one pattern i like in general for optional callbacks is to set a noop default value:
```
  public static defaultProps: Partial<IProps> = {
    tagCleared: () => {}
  };
```
then you don't have to check if it exists everywhere you want to call it. Doesn't buy much in a simple component like this, but I've found it's a good pattern for larger components to not litter these if checks. Thoughts?